lang/c++/impl/Compiler.cc

```diff
@@ -229,9 +231,13 @@ static GenericDatum makeGenericDatum(NodePtr n,
         assertType(e, json::etLong);
         return GenericDatum(e.longValue());
     case AVRO_FLOAT:
+        if (dt == json::etLong)
```

Comment:
In order to make it consistent with the rest of C++ code, can you add braces?

     if (dt == json::etLong) {
         return GenericDatum(static_cast<float>(e.longValue()));
     }